Dockerfile

```diff
@@ -102,12 +103,10 @@ ENV TZ=${TZ} \
     ${ADD_DEB_PACKAGES}"
 
 WORKDIR /pygeoapi
-ADD . /pygeoapi
 
 # Install operating system dependencies
 RUN \
     apt-get update -y \
-    && apt-get upgrade -y \
```

Comment:
I think in order to get the latest versions of libraries in containers, the idea is to use the latest base image directly. If updates happen here, then both the old and the new library are included in the image which can make it a lot larger depending on how many updates happen here.

Also when image caches are used, you never know which library versions are being used because an image built with this instruction can be cached for a long time.

There's also the issue of reproducibility, because when an updated package is available, it will be used on the next image build. So it can happen that we verify that pygeoapi is working fine with a certain set of libraries, then we need to release a hotfix, but we can't because a new library version is pulled in here which could create an incompatibility.
This is something that can be avoided by selecting the base image version manually at a time of our choosing.

I realize that going with using the latest image and also doing the upgrades here is a feasible path forward, so i'd also be fine with that if there are good reasons for that.

Reply:
I think we should be using the latest official image available and not run apt upgrade ourselves.

Reply:
@kalxas I have the impression that taking only the latest official image doesn't preserve us to get vulnerable python packages. This might be fixed with the apt upgrade to get the remediated version of a package if any

Reply:
I think the onus is on the base image to be updated accordingly.